From 46fea6639d4e769a888a83bf99d7789ddcf92b5f Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.s.prusty@intel.com>
Date: Mon, 25 Aug 2014 15:45:03 +0530
Subject: [PATCH 07/12] Audio: Delay enabling DAPM widgets during recovery
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

This is to ensure that the previously running streams are
closed successfully before the DAPM widgets are re-enabled.
In the absence of this delay, the widgets could get enabled
for streams that are in the process of getting closed.

Change-Id: I2098b3a3a332657cbfe49152d0301be759c31719
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
---
 sound/soc/intel/sst/sst_pvt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/intel/sst/sst_pvt.c b/sound/soc/intel/sst/sst_pvt.c
index 5970508..23fb319 100644
--- a/sound/soc/intel/sst/sst_pvt.c
+++ b/sound/soc/intel/sst/sst_pvt.c
@@ -435,6 +435,10 @@ void sst_do_recovery_mrfld(struct intel_sst_drv *sst)
 	sst_stream_recovery(sst);
 	mutex_unlock(&sst->sst_lock);
 
+	/* Delay is to ensure that the stream is closed before
+	 * powering on DAPM widget
+	 */
+	usleep_range(10000, 12000);
 	reset_dapm = false;
 	sst_platform_cb(&cb_params);
 }
-- 
1.8.3.2

